### DISCUSSION TITLE
NSPanel_HA_Blueprint 4.2.3 Version mismatch

### DISCUSSION THREAD

**Discussion by theoml** [Blackymas/NSPanel_HA_Blueprint#1682] [Q&A]
Hi,
yesterday installed the new Firmware and reimported the new Version of Blueprint 4.2.3. Did the update as mentioned (I did that successful befor with earlier Versions). My Display shows ESPHome Ver 4.2.3, TFT 4.2.3 but Blueprintversion is pending.(Log) and also not shown at the NSPanel.
The Blueprint at Panel config shows definive 4.2.3. 
Did the reimport of the Blueprint multiple times with no change.
The Display now shows the HA Icon in red

Where ist the fault?

Regards
Reiner

HA is installed on RPI 4, 2GB everything is actual

**Comment by edwardtfn** [comment]
I believe this was fixed with v4.2.4, but if you still having this issue after updating your system, could you please report it as a bug in the [Issues](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues) area?

**Comment by theoml** [comment]
Hi Edward!

 

Did not get the Message about 4.2.4.

Just installed and everything is fine again.

 

Why 3 Updates in such a short time.

 

Regards
Reiner

 

Von: Edward Firmo ***@***.***> 
Gesendet: Donnerstag, 18. Januar 2024 08:11
An: Blackymas/NSPanel_HA_Blueprint ***@***.***>
Cc: theoml ***@***.***>; Author ***@***.***>
Betreff: Re: [Blackymas/NSPanel_HA_Blueprint] NSPanel_HA_Blueprint 4.2.3 Version mismatch (Discussion #1682)

 

I believe this was fixed with v4.2.4, but if you still having this issue after updating your system, could you please report it as a bug in the Issues <https://github.com/Blackymas/NSPanel_HA_Blueprint/issues>  area?

**Reply by edwardtfn** [reply]
> Why 3 Updates in such a short time.

This is definitely something we always try to avoid, however, we got in a sequence where fixing an error was breaking something else for other people.
On v4.2 we introduced a common engine to fetch entities attributes (icons, colors, names, etc.), in contrast with the previous versions where different code was doing the same thing in multiples parts of the blueprint. This brings consistency and in the long term, stability to the system, but makes that single component a very critical piece and any bug there becomes critical (which means it prevents someone to use it's panel or a feature completely).
Every time we have a fix to a critical bug, we starts with the dilemma of releasing as soon as possible, and so relief the ones feeling the issue, or accumulate with other fixes and/or to have more time for tests. And typically I take a decision also based on my availability to support, as this I have a job, a family, friends, etc. outside this project and try to fit it better. Other collaborators have been quite busy with their things also, so I'm almost the sole developer for the past few versions, with rare exceptions when someone from the community fixes things here and there, so if the error doesn't happens to me, there's a big chance it will hit a released version.
This brings to another issue, that we don't have many people testing things on `dev`, so unfortunately many bugs aren't visible until they reach a wider audience.

But we also try to make updates "optional", so if everything is working for you, there's no need to follow every patch. Usually the patch release note have all the details of the changes and components affected.

**Comment by theoml** [comment]
Thanks for your explanation.I thought that there is something wrong with that Version and reimported the blueprint a few times, but this morning I thought its the time to ask.Thanks a lot for your unusual work.RegardsReiner Am 18.01.2024 10:05 schrieb Edward Firmo ***@***.***>:

Why 3 Updates in such a short time.

This is definitely something we always try to avoid, however, we got in a sequence where fixing an error was breaking something else for other people.
On v4.2 we introduced a common engine to fetch entities attributes (icons, colors, names, etc.), in contrast with the previous versions where different code was doing the same thing in multiples parts of the blueprint. This brings consistency and in the long term, stability to the system, but makes that single component a very critical piece and any bug there becomes critical (which means it prevents someone to use it's panel or a feature completely).
Every time we have a fix to a critical bug, we starts with the dilemma of releasing as soon as possible, and so relief the ones feeling the issue, or accumulate with other fixes and/or to have more time for tests. And typically I take a decision also based on my availability to support, as this I have a job, a family, friends, etc. outside this project and try to fit it better. Other collaborators have been quite busy with their things also, so I'm almost the sole developer for the past few versions, with rare exceptions when someone from the community fixes things here and there, so if the error doesn't happens to me, there's a big chance it will hit a released version.
This brings to another issue, that we don't have many people testing things on dev, so unfortunately many bugs aren't visible until they reach a wider audience.
But we also try to make updates "optional", so if everything is working for you, there's no need to follow every patch. Usually the patch release note have all the details of the changes and components affected.
